gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -399,22 +399,35 @@ def prop_defined(name, cls, rules={})
     T::Configuration.without_ruby_warnings do
       if !rules[:immutable]
         if method(:prop_set).owner != T::Props::Decorator
-          @class.send(:define_method, "#{name}=") do |val|
-            T.unsafe(self.class).decorator.prop_set(self, name, val, rules)
-          end
+          @class.class_eval <<~SETTER
+            def #{name}=(val)
+              rules = self.class.decorator.prop_rules(:#{name})
+              self.class.decorator.prop_set(self, :#{name}, val, rules)
```

Comment:
Since we have to fetch the decorator, can we at least factor out `self.class.decorator` into a local?  Here and elsewhere in this patch.